### PR TITLE
openssl: switch to linux-x86 and linux-x86_64 targets

### LINKED ISSUE #71455: `pkgsCross.musl64.openssl` stops compiling
**Describe the bug**
As of 21st October 2019, `pkgsCross.musl64.openssl` stops building on `release-19.09` branch.

**To Reproduce**
Build `pkgsCross.musl64.openssl`.
The build will fail because the necessary assembly code is not emitted.
The missing assembly code appears to do with Intel AES-NI extensions.

**Expected behavior**
Cross compiling openssl to Linux MUSL target should succeed.

**Additional context**
It appears that it is an upstream issue when `openssl` is configured for `linux-generic*` targets.
The immediate remedy would be building `openssl` with `linux-x86` for `x86-linux` systems and `linux-x86_64` for `x86_64-linux` systems. Currently I am studying the difference between the `linux-generic*`, `linux-x86` and `linux-x86_64` variants. If the difference is not that very significant, I guess we can apply this remedy.

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.6, NixOS, 19.09.840.8bf142e001b (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3`
 - channels(dingxf): `"home-manager-19.09"`
 - channels(root): `"nixos-19.09.840.8bf142e001b, nixos-hardware"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by dingxiangfei2009**
###### Motivation for this change
Fix #71455 
This PR switches Linux builds to target `linux-x86_64` and `linux-x86` instead of `linux-generic*`, while the upstream resolve the cross-compilation issue.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @peti

### CODE REVIEW

**Comment by FRidh** [comment]
cc @dtzWill @matthewbauer for cross-compiling / musl

File: pkgs/development/libraries/openssl/default.nix [hunk 1]
```diff
@@ -44,10 +44,12 @@ let
     # TODO(@Ericson2314): Improve with mass rebuild
     configurePlatforms = [];
     configureScript = {
-        x86_64-darwin  = "./Configure darwin64-x86_64-cc";
-        x86_64-solaris = "./Configure solaris64-x86_64-gcc";
         armv6l-linux = "./Configure linux-armv4 -march=armv6";
         armv7l-linux = "./Configure linux-armv4 -march=armv7-a";
+        x86-linux = "./Configure linux-x86";
```
Comment (veprbl) [inline pkgs/development/libraries/openssl/default.nix]: Is `"x86-linux"` a valid value for `stdenv.hostPlatform.system`?

**Comment by petabyteboy** [comment]
I can confirm that this change fixes the compilation of pkgsCross.musl64.openssl.
Can you please adjust the commit message to adhere to the contributing guidelines by prefixing it with "openssl: "?

File: pkgs/development/libraries/openssl/default.nix (hunk 1 above)
Comment (dingxiangfei2009) [reply pkgs/development/libraries/openssl/default.nix]: I am so damned. It should be `i686-linux`, sorry.

**Comment by dingxiangfei2009** [comment]
@petabyteboy Yes, definitely. Sorry for the inconvenience.

**Comment by petabyteboy** [comment]
Looks good to me now

**Review by matthewbauer — APPROVED** [review]

**Comment by FRidh** [comment]
Ugh, wrong branch. Please check that as well when reviewing (and merging...)!

**Comment by FRidh** [comment]
703e44675c5e62b0dbbd37b76153383fc0b5ee47 is now in staging

**Comment by FRidh** [comment]
This broke pkgs.i686Linux.openssl_1_0_2, https://github.com/NixOS/nixpkgs/issues/71786

**Pull request merged**